packages/filebrowser/src/listing.ts

```diff
@@ -1142,11 +1136,54 @@ export class DirListing extends Widget {
       }
     }
 
-    // Write to DOM
+    // Compose the grid columns.
+    const columnOrder = [
+      'jp-filebrowser-checkbox',
+      'jp-filebrowser-name',
+      'jp-filebrowser-resize-name-modified',
+      'jp-filebrowser-modified',
+      'jp-filebrowser-resize-modified-size',
+      'jp-filebrowser-filesize'
+    ];
+    let columnWidths: Record<string, string> = {
+      'jp-filebrowser-checkbox': '24px',
+      'jp-filebrowser-name': 'auto',
+      'jp-filebrowser-resize-name-modified': '5px',
+      'jp-filebrowser-modified': 'auto',
+      'jp-filebrowser-resize-modified-size': '5px',
+      'jp-filebrowser-filesize': 'auto'
+    };
+
     for (const column of visibleColumns) {
       const size = this._columnSizes[column.id];
-      column.element.style.width = size === null ? '' : size + 'px';
+      const newSize = size === null ? 'auto' : size + 'px';
+      const colName = column.className.replace(
+        RegExp(/^jp-id-/),
+        'jp-filebrowser-'
+      );
+      columnWidths[colName] = newSize;
     }
+
+    // Update the header's template columns.
+    const newGridTemplateColumns = columnOrder
+      .map(col => {
+        // Allow the columns containing file information to grow.
+        const newWidth = col.match(/^jp-filebrowser-(name|modified|filesize)$/)
```

Comment:
We should move it to specification of columns, rather than hard-code names here. The common request is to allow adding columns in extensions, but hard-coding logic based on column name makes it harder.